From 5193198dc9caabedfdd801a94bf936c69758aa14 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 19 Aug 2015 13:59:40 +0200
Subject: [PATCH] network/ifup.sh:do_static(): error out, if interface could
 not be brought up

(cherry picked from commit 77f46adf5e5ab1f6da2e459bb55435d4b70842c5)
---
 modules.d/40network/ifup.sh | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/modules.d/40network/ifup.sh b/modules.d/40network/ifup.sh
index 7da9171..69a4567 100755
--- a/modules.d/40network/ifup.sh
+++ b/modules.d/40network/ifup.sh
@@ -150,7 +150,11 @@ do_ipv6auto() {
 do_static() {
     strglobin $ip '*:*:*' && load_ipv6
 
-    linkup $netif
+    if ! linkup $netif; then
+        warn "Could bring interface $netif up!"
+        return 1
+    fi
+
     [ -n "$macaddr" ] && ip link set address $macaddr dev $netif
     [ -n "$mtu" ] && ip link set mtu $mtu dev $netif
     if strglobin $ip '*:*:*'; then
